src/main/java/de/tum/in/www1/artemis/service/metis/conversation/auth/ConversationAuthorizationService.java

```diff
@@ -0,0 +1,50 @@
+package de.tum.in.www1.artemis.service.metis.conversation.auth;
+
+import java.util.Objects;
+
+import javax.persistence.Persistence;
+
+import org.springframework.stereotype.Service;
+
+import de.tum.in.www1.artemis.domain.User;
+import de.tum.in.www1.artemis.repository.UserRepository;
+import de.tum.in.www1.artemis.repository.metis.ConversationParticipantRepository;
+import de.tum.in.www1.artemis.service.AuthorizationCheckService;
+
+@Service
+public class ConversationAuthorizationService {
+
+    protected final ConversationParticipantRepository conversationParticipantRepository;
+
+    protected final UserRepository userRepository;
+
+    protected final AuthorizationCheckService authorizationCheckService;
+
+    protected ConversationAuthorizationService(ConversationParticipantRepository conversationParticipantRepository, UserRepository userRepository,
+            AuthorizationCheckService authorizationCheckService) {
+        this.conversationParticipantRepository = conversationParticipantRepository;
+        this.userRepository = userRepository;
+        this.authorizationCheckService = authorizationCheckService;
+    }
+
+    /**
+     * Returns a {@link User} object with authorities and groups loaded.
+     *
+     * @param user the {@link User} object to check for loaded authorities and groups
+     * @return the {@link User} object, potentially after loading the authorities and groups
+     * @throws IllegalArgumentException if the user parameter is null
+     */
+    protected User getUserIfNecessary(User user) {
```

Comment:
Shouldn't this maybe be in the User/Auth Service? I understand it is a helper method, but it does not perform anything specific to the Conversation Service.

Reply:
The corresponding method in the auth service is private and is slightly different as it assumes if user is null then the currently requesting user is meant. I would rather not introduce another public method for my case here.